### ISSUE TITLE
Camera Framerate on 32GB Mi2s

### ISSUE THREAD

**Issue by shrthndin** [mitwo-dev/android_device_xiaomi_aries#20]
Is it possible to increase the framerate range of preview images in the Mi2s 32gb driver to support 0-120fps? The camera driver is locked to only support 10-25fps(PAL?). The Sony sensor supports 1080p/30-60fps. I need to lock the framerate to the intended setting for an app I've been working on.

**Comment by Ryotsuke** [comment]
I noticed I can't record video over 22fps like 24fps or 30fps on 32gb mi2s
This is bad too

**Comment by shrthndin** [comment]
Yes, unless you're in hdr mode it averages at 22.5. This is from the preview range being clamped at 10-25fps. The driver has to allow the range from 0-120fps to be able to lock the framerate.

**Comment by guari** [comment]
Is there some news about this bug? If the sensor is the Exmor RS IMX135, according to Sony spec the driver should allow 1080p@60fps in normal mode and 1080p@30fps in HDR mode.

**Comment by ivan19871002** [comment]
i think the snapdragon 600 does not support.

ÔÚ 2013Äê12ÔÂ21ÈÕ 04:19:06
"notifications"notifications@github.com Ð´µÀ£º

Is there some news about this bug? If the sensor is the Exmor RS IMX135, according to Sony spec the driver should allow 1080p@60fps in normal mode and 1080p@30fps in HDR mode. 

&mdash;

**Comment by ivan19871002** [comment]
600 is support 1080p@30fps 720p@60fps   video record

**Comment by guari** [comment]
@ivan19871002 You're right, I see that Qualcomm corrected its documents from the last time I read, now it's 1080p@30fps and 720p@60fps like you said, but in mi2s with 13mp sensor, video recording in normal mode is locked at 1080p@22.5fps (as max framerate in VFR, the real fps might be also a lower value).

This results in 1080p video recording affected by noticeable lags and wobbling effects that makes the overall quality far from IMX135 capabilities, is there some other bottle-neck?

(as reported by many users mi2s 16gb/8MP do not suffer this issue)

**Comment by M1cha** [comment]
do other phones with a imx135 sensor support higher framerates?

**Comment by guari** [comment]
As far as I know, Xperia Z, Galaxy S4, LG G2/G3 etc. should record at least at 1080p@30fps and 720p@60fps (indipendently from HDR processing).

It can be checked by downloading a sample video from youtube and by watching its spec.

(eg. for xperia z, this doesn't seems to use HDR and it's 30fps: http://www.youtube.com/watch?v=mfAWQKyqHYs, while a Mi2S 13MP sample is averaged at 22fps: http://www.youtube.com/watch?v=dYmEjBiehv4)

**Comment by M1cha** [comment]
you should try the libchromatix_imx135_*.so files from these devices(you need a cam v1 device).
Also YouTube converts all videos which means it's not reliable for comparison.

**Comment by guari** [comment]
thanks @M1cha I'll take a look at it.
I'm afraid that most of them uses a camera HAL v3 instead of the old v1.. but need to check yet.

yes, youtube converts the video, but if you upload and download a mp4 @1080p, the framerate and the overall quality of the video remains pretty the same, they compress the video to fit the standard resolutions or if the bitrate is too much high for streaming.

**Comment by M1cha** [comment]
v3? I know about v1 and v2 only :/
also, we could upgrade our libs :P

**Comment by guari** [comment]
If I've not misunderstand you.. :) I thought you're talking about https://source.android.com/devices/camera/camera.html (where [v3.1](https://source.android.com/devices/camera/camera3.html) is latest shared lib interface)

(but currently I've not great experience in android portings, never played with it, so let me know if I'm saying stupid things.. :D)

**Comment by M1cha** [comment]
I'm talking about qcom's camera architecture :)

**Comment by guari** [comment]
yup thanks for the clarification :P
I'd say I need a little more search to understand well how things works..
